lib/src/facts/windows/dmi_resolver.cc

```diff
@@ -18,11 +18,19 @@ namespace facter { namespace facts { namespace windows {
         data result;
 
         auto vals = _wmi->query(wmi::computersystemproduct, {wmi::name});
-        result.product_name = wmi::get(vals, wmi::name);
+        if (vals.empty()) {
+            LOG_DEBUG("failure resolving dmi::product::name fact, wmi could not resolve %1%::%2%", wmi::computersystemproduct, wmi::name);
```

Comment:
`dmi::product::name` might be confusing as it's not a fact name.  Perhaps something like `LOG_DEBUG("WMI query returned no results for %1% with value %2%.", wmi::computersystemproduct, wmi::name);`?